hooks/pre-command

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -euo pipefail
+
+_SOURCE="${BASH_SOURCE[0]}"
+[ -z "${_SOURCE:-}" ] && _SOURCE="${0}"
+BASEDIR="$( cd "$( dirname "${_SOURCE}" )" && cd .. && pwd )"
+
+if [[ -z "${BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_AUDIENCE:-}" ]]; then
+  echo "🚨 Missing 'audience' plugin configuration"
+  exit 1
+fi
+
+if [[ -z "${BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_SERVICE_ACCOUNT:-}" ]]; then
+  echo "🚨 Missing 'service-account' plugin configuration"
+  exit 1
+fi
+
+echo "--- :buildkite: Requesting OIDC token from Buildkite"
+
+buildkite-agent oidc request-token --audience "$BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_AUDIENCE" > "$BASEDIR"/token.json
+
+echo "--- :gcp: Configuring Google Cloud credentials"
+
+cat << JSON > "$BASEDIR"/credentials.json
```

Comment:
I wonder if we should have a `pre-exit` hook to remove this file after the command?

Reply:
Updated to clean up credentials on `pre-exit`